src/Compiler/VMCode.idr

```diff
@@ -127,19 +165,18 @@ toVM t res (AOp fc _ op args)
     = [OP res op (map toReg args)]
 toVM t res (AExtPrim fc _ p args)
     = [EXTPRIM res p (map toReg args)]
+toVM t res (AConCase fc (ALocal scr) [MkAConAlt n ci mt args code] Nothing) -- exactly one alternative, so skip matching
+    = let body = toVM t res code
+          used = foldMap collectUsed body
+       in projectArgs scr 0 used args ++ body
```

Comment:
Such an optimization is also possible for cases with zero alternatives and just a single default value as well as for matches on constants. Or are all of these already optimized in some earlier IR (in which case this should probably be reflected by the types)? I'm asking this because the refactored JS backend in #1609 performs the same optimizations.

Reply:
I don't know if that happens, but I've only optimised this because of how common record projections are. I don't expect it to be very common to have no alternatives and just a default, but that can be easily optimised at the `CExp` level